armory/datasets/standard/resisc45/resisc45.py

```diff
@@ -0,0 +1,134 @@
+   pages={1865-1883}
+}"""
+
+_DESCRIPTION = """\
+RESISC45 dataset is a publicly available benchmark for Remote Sensing Image
+Scene Classification (RESISC), created by Northwestern Polytechnical University
+(NWPU). This dataset contains 31,500 images, covering 45 scene classes with 700
+images in each class. This is a variation of that dataset with 500/100/100
+train/validate/test splits for each class.
+"""
+
+_LABELS = [
+    "airplane",
+    "airport",
+    "baseball_diamond",
+    "basketball_court",
+    "beach",
+    "bridge",
+    "chaparral",
+    "church",
+    "circular_farmland",
+    "cloud",
+    "commercial_area",
+    "dense_residential",
+    "desert",
+    "forest",
+    "freeway",
+    "golf_course",
+    "ground_track_field",
+    "harbor",
+    "industrial_area",
+    "intersection",
+    "island",
+    "lake",
+    "meadow",
+    "medium_residential",
+    "mobile_home_park",
+    "mountain",
+    "overpass",
+    "palace",
+    "parking_lot",
+    "railway",
+    "railway_station",
+    "rectangular_farmland",
+    "river",
+    "roundabout",
+    "runway",
+    "sea_ice",
+    "ship",
+    "snowberg",
+    "sparse_residential",
+    "stadium",
+    "storage_tank",
+    "tennis_court",
+    "terrace",
+    "thermal_power_station",
+    "wetland",
+]
+
+_HOMEPAGE = "http://www.escience.cn/people/JunweiHan/NWPU-RESISC45.html"
+
+_URL = "https://armory-public-data.s3.us-east-2.amazonaws.com/resisc45/NWPU-RESISC45.tar.gz"
+
+
+class Resisc45(tfds.core.GeneratorBasedBuilder):
+    """NWPU Remote Sensing Image Scene Classification (RESISC) Dataset."""
+
+    VERSION = tfds.core.Version("3.0.0")
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict(
+                {
+                    "image": tfds.features.Image(shape=[256, 256, 3]),
+                    "label": tfds.features.ClassLabel(names=_LABELS),
+                    "filename": tfds.features.Text(),
+                }
+            ),
+            supervised_keys=("image", "label"),
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        path = dl_manager.download_and_extract(_URL)
+        return {
+            split: self._generate_examples(path, split)
+            for split in ["train", "validation", "test"]
+        }
+
+    def _generate_examples(self, path, split):
+        """Yields examples."""
+        for label in tf.io.gfile.listdir(f"{path}/NWPU-RESISC45"):
+            for idx, filename in enumerate(
+                tf.io.gfile.glob(f"{path}/NWPU-RESISC45/{label}/*.jpg")
+            ):
+                example = {
+                    "image": filename,
+                    "label": label,
+                    "filename": os.path.basename(filename),
+                }
+                if idx <= 500:
```

Comment:
There are three issues here. 

1. Currently, I'm not sure if glob will sort, so I would want to do a `sorted(enumerate(...` with your method (even if it does, that will make it explicit). 
2. Then you need to change the `<=` to `<`, since `enumerate` is 0-indexed but the dataset items are 1-indexed.
3. Since we are no longer enforcing explicit names, I would want to raise an error if `idx >= 700`. That would indicate that there is an erroneous file with a `.jpg` ending.

Alternatively, you could raise an error if `filename != f"{label}_{idx+1:03}.jpg"`. That would make it very clear what is going on.